Library/Homebrew/extend/os/mac/utils/bottles.rb

```diff
@@ -11,6 +11,11 @@ def tag(tag = nil)
 
           super
         end
+      
+        sig {returns(T::Boolean) }
+        def on_macos?
+          true
+        end
```

Comment:
Instead of defining an `on_macos?` method (which is essentially the same thing as checking for `OS.mac?`), instead, implement each method (e.g. `binaries_need_relocation?` as needed inside each of `extend/os/{mac,linux}/utils/bottles.rb`.